Installation issue: paraview@master Could NOT find catalyst (missing: catalyst_DIR)

<!-- Thanks for taking the time to report this build failure. To proceed with the report please:

1. Title the issue "Installation issue: <name-of-the-package>".
2. Provide the information required below.

We encourage you to try, as much as possible, to reduce your problem to the minimal example that still reproduces the issue. That would help us a lot in fixing it quickly and effectively! -->

### Steps to reproduce the issue

@mathstuf I'm guessing the recent addition of `VTK_MODULE_USE_EXTERNAL_ParaView_vtkcatalyst` is getting turned on in spack.

<!-- Fill in the exact spec you are trying to build and the relevant part of the error message -->
```console
$ spack install paraview@master
...
      235    -- Could NOT find catalyst (missing: catalyst_DIR)
  >> 236    CMake Error at VTK/CMake/vtkModule.cmake:4366 (message):
     237      Could not find the catalyst external dependency.
     238    Call Stack (most recent call first):
     239      VTK/CMake/vtkModule.cmake:4960 (vtk_module_find_package)
     240      VTK/CMake/vtkModule.cmake:4831 (vtk_module_third_party_external)
     241      ThirdParty/catalyst/CMakeLists.txt:31 (vtk_module_third_party)
```

### Information on your system

<!-- Please include the output of `spack debug report` -->

```console
* **Spack:** 0.16.2-3651-25c09e7a56
* **Python:** 3.6.3
* **Platform:** linux-rhel7-broadwell
* **Concretizer:** original
```

<!-- If you have any relevant configuration detail (custom `packages.yaml` or `modules.yaml`, etc.) you can add that here as well. -->

### Additional information

<!-- Please upload the following files. They should be present in the stage directory of the failing build. Also upload any config.log or similar file if one exists.
* [spack-build-out.txt]()
* [spack-build-env.txt]() -->

<!-- Some packages have maintainers who have volunteered to debug build failures. Run `spack maintainers <name-of-the-package>` and @mention them here if they exist. -->

### General information

<!-- These boxes can be checked by replacing [ ] with [x] or by clicking them after submitting the issue. -->
- [x] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [ ] I have run `spack maintainers <name-of-the-package>` and @mentioned any maintainers
- [ ] I have uploaded the build log and environment files
- [x] I have searched the issues of this repo and believe this is not a duplicate

--- a/var/spack/repos/builtin/packages/paraview/package.py
+++ b/var/spack/repos/builtin/packages/paraview/package.py
@@ -273,6 +273,9 @@ class Paraview(CMakePackage, CudaPackage):
             '-DBUILD_TESTING:BOOL=OFF',
             '-DOpenGL_GL_PREFERENCE:STRING=LEGACY']
 
+        if spec.satisfies('@5.10:'):
+            cmake_args.append('-DVTK_MODULE_USE_EXTERNAL_ParaView_vtkcatalyst:BOOL=OFF')
+
         if spec.satisfies('@:5.7') and spec['cmake'].satisfies('@3.17:'):
             cmake_args.append('-DFPHSA_NAME_MISMATCHED:BOOL=ON')
 

